### PR TITLE
Read/write module EEPROM via SDK provided sys fs

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
#### Why I did it

Currently, implementation:

- Reading EEPROM: via ethtool
- Writing EEPROM: via mlxreg tool.

The problem is that spawning new process of ethtool/mlxreg is very slow. To optimize the performance, this PR will use SDK provided sys fs for EEPROM reading/writing.

#### How I did it

Replace the logic of `sfp.read_eeprom` and `sfp.write_eeprom` by SDK sys fs.

#### How to verify it

Manual test
Unit test cases (new test case added)
sonic-mgmt regression

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205

### CODE REVIEW

**Comment by fastiuk** [comment]
I am reviewing your PR. Will provide feedback in the nearest hour

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 1]
```diff
@@ -344,80 +309,48 @@ def get_presence(self):
...
-        # recalculate offset and page. Use 'ethtool' if there is no need to read vendor pages
-        if offset < SFP_VENDOR_PAGE_START:
-            return self._read_eeprom_specific_bytes(offset, num_bytes)
-        else:
-            page = (offset - SFP_PAGE_SIZE) // SFP_UPPER_PAGE_OFFSET + 1
-            # calculate offset per page
-            device_address = (offset - SFP_PAGE_SIZE) % SFP_UPPER_PAGE_OFFSET + SFP_UPPER_PAGE_OFFSET
+        return self._read_eeprom(offset, num_bytes)
+
+    def _read_eeprom(self, offset, num_bytes, log_on_error=True):
```
Comment (fastiuk) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:325]: I think this is not the best place to put EEPROM API.
This API can be reusable for other EEPROM devices not only for SFP.
I would put this API into another module like eeprom.py and use it here.

**Review by fastiuk — CHANGES_REQUESTED** [review]

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 2]
```diff
@@ -344,80 +309,48 @@ def get_presence(self):
...
+                f.seek(page_offset)
+                content = f.read(num_bytes)
+                if ctypes.get_errno() != 0:
+                    logger.log_error('Failed to read sfp={} EEPROM page={}, page_offset={}, size={}, offset={}, errno = {}'.format(
+                        self.sdk_index, page, page_offset, num_bytes, offset, os.strerror(ctypes.get_errno())))
+                    return None
+                return bytearray(content)
+        except (OSError, IOError) as e:
+            if log_on_error:
+                logger.log_error('Failed to read sfp={} EEPROM page={}, page_offset={}, size={}, offset={}, exception = {}'.format(
```
[the remark this answers is not in the document]
Comment (Junchao-Mellanox) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:351]: 1. will fix
2. logger is an instance of sonic_py_common.logger, it has no method "exception"

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 1 above)
Comment (Junchao-Mellanox) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:325]: It is only for SFP EEPROM reading. eeprom.py is for system EEPROM.

**Review by fastiuk — APPROVED** [review]

**Pull request closed without merging**